python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -301,5 +537,221 @@ def run(
+
+        columns_target = {
+            target.column_coord1: (
+                ref.column_coord1, target.column_coord1, coord1_target_err, False,
+            ),
+            target.column_coord2: (
+                ref.column_coord2, target.column_coord2, coord2_target_err, False,
+            ),
+            column_dist: (column_dummy, column_dist, column_dist_err, False),
+        }
+
+        band_prev = None
+        for idx_band, (band, config_flux) in enumerate(config.columns_flux.items()):
+            mag_ref = -2.5 * np.log10(cat_ref[config_flux.column_ref_flux]) + config.mag_zeropoint_ref
+            if idx_band > 0:
+                cat_ref[column_color_temp] = cat_ref[column_mag_temp] - mag_ref
+            cat_ref[column_mag_temp] = mag_ref
+
+            flux_err_frac = [None]*n_models
+            mag_model = [None]*n_models
+
+            select_ref_bins = [select_ref & (mag_ref > mag_lo) & (mag_ref < mag_hi)
+                               for idx_bin, (mag_lo, mag_hi) in enumerate(bins_mag)]
+
+            # Iterate over multiple models, compute their mags and colours (if there's a previous band)
+            for idx_model in range(n_models):
+                column_target_flux = config_flux.columns_target_flux[idx_model]
+                column_target_flux_err = config_flux.columns_target_flux_err[idx_model]
+
+                flux_target = cat_target[column_target_flux]
+                mag_target = -2.5*np.log10(flux_target) + config.mag_zeropoint_target
+                if config.mag_ceiling_target is not None:
+                    mag_target[mag_target > config.mag_ceiling_target] = config.mag_ceiling_target
+                mag_model[idx_model] = mag_target
+
+                # These are needed for computing magnitude/color "errors" (which are a sketchy concept)
+                flux_err_frac[idx_model] = cat_target[column_target_flux_err]/flux_target
+                # Keep these mags tabulated for convenience
+                column_mag_temp_model = f'{column_mag_temp}{idx_model}'
+                cat_target[column_mag_temp_model] = mag_target
+
+                columns_target[f'flux_{column_target_flux}'] = (
+                    config_flux.column_ref_flux,
+                    column_target_flux,
+                    column_target_flux_err,
+                    True,
+                )
+                # Note: magnitude errors are generally problematic and not worth aggregating
+                columns_target[f'mag_{column_target_flux}'] = (
+                    column_mag_temp, column_mag_temp_model, None, False,
+                )
+
+                if idx_band > 0:
+                    column_color_temp_model = f'{column_color_temp}{idx_model}'
+                    column_color_err_temp_model = f'{column_color_err_temp}{idx_model}'
+
+                    # e.g. if order is ugrizy, first color will be u - g
+                    cat_target[column_color_temp_model] = mag_prev[idx_model] - mag_model[idx_model]
```

Comment:
Are they guaranteed to be in order?

Reply:
They're in config.columns_flux order, so the user can specify the N-1 colours they want. I suppose I could output all N*(N-1)/2 unique colours instead, since that's "only" 15 instead of 5. I'll leave that for a future hypothetical feature request.